scripts/factory-hooks/post-site-install/post-site-install.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+#
+# Factory Hook: post-site-install
+#
+# This is used so that an ACSF site install is identical to the local BLT site
+# install, with the environment, site, and uri CLI runtime arguments overriding
+# all other configuration.
+#
+# Usage: SCRIPTNAME site env db-role domain
+
+# Exit immediately on error and enable verbose log output.
+set -ev
+
+# Map the script inputs to convenient names:
+# Acquia Hosting sitegroup (application) and environment.
+sitegroup="$1"
+env="$2"
+# Database role. This is a truly unique identifier for an ACSF site and is e.g.
+# part of the public files path.
+db_role="$3"
+# Internal (Acquia managed) domain name of the website. (No public domain name
+# is assigned yet, immediately after installation.) The first part is a name
+# that is unique per installed site. A small but significant difference with
+# $db_role: if a site gets deleted and reinstalled with the same name, it gets
+# a different $db_role.
+internal_domain="$4"
+# To get only the site name in ${name[0]}:
+IFS='.' read -a name <<< $internal_domain
+
+# BLT executable:
+blt="/mnt/www/html/$sitegroup.$env/vendor/acquia/blt/bin/blt"
+
+# Create and set Drush cache to unique local temporary storage per site.
+# This approach isolates drush processes to completely avoid race conditions
+# that persist after initial attempts at addressing in BLT: https://github.com/acquia/blt/pull/2922
+cache_dir=`/usr/bin/env php /mnt/www/html/$sitegroup.$env/vendor/acquia/blt/scripts/blt/drush/cache.php $sitegroup $env $internal_domain`
+
+# Print to Site Factory SiteInstall task log.
+echo "Executing: DRUSH_PATHS_CACHE_DIRECTORY=$cache_dir $blt drupal:update --environment=$env --site=${name[0]} --define drush.uri=$internal_domain --verbose --no-interaction"
+
+# Execute the updates.
+DRUSH_PATHS_CACHE_DIRECTORY="$cache_dir" $blt drupal:update --environment=$env --site=${name[0]} --define drush.uri=$internal_domain --verbose --no-interaction
+result=$?
+
+# Clean up the drush cache directory.
+rm -rf "$cache_dir"
+
+set +v
+
+# Exit with the status of the BLT commmand. If the exit status is non-zero,
+# Site Factory will send a notification of a partiolly failed install and will
+# stop executing any further post-site-install hook scripts that would be in
+# this directory (and get executed in alphabetical order).
+exit $result
```

Comment:
There are a few changes I couldn't resist making relative to db-update.sh, and/or didn't know whether I should make, it's fairly subjective:
* use snake_case for bash variables ($cache_dir not $cacheDir) because I've never seen camelCase in bash before.
* convention seems to be to uppercase bash variables, actually. But I don't know if I should - I just kept it equal to db-update.sh
* quote all variables? Officially yes. But none of the inputs ever contain spaces and things become less readable and db-update.sh does _mostly not_ do it (though it isn't consistent). I did quote $cache_dir because I feel icky _not_ doing it with a variable that is used in a 'rm -rf' command. (And then I'm doing it consistently throughout the script.)